ios/browser/api/web/ui/chrome_webview.mm

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/ios/browser/api/web/ui/chrome_webview.h"
+
+#include "base/memory/weak_ptr.h"
+#include "base/strings/sys_string_conversions.h"
+#include "ios/chrome/browser/shared/model/application_context/application_context.h"
+#include "ios/chrome/browser/shared/model/browser_state/chrome_browser_state.h"
+#include "ios/chrome/browser/shared/model/browser_state/chrome_browser_state_manager.h"
+#include "ios/components/webui/web_ui_url_constants.h"
+#import "ios/web/js_messaging/web_view_web_state_map.h"
+#import "ios/web/public/browser_state.h"
+#import "ios/web/public/navigation/navigation_manager.h"
+#include "ios/web/public/thread/web_thread.h"
+#import "ios/web/public/web_state.h"
+#import "ios/web/public/web_state_delegate_bridge.h"
+#import "ios/web/public/web_state_observer_bridge.h"
+#import "ios/web/public/webui/web_ui_ios.h"
+#import "ios/web/web_state/ui/crw_web_controller.h"
+#import "ios/web/web_state/ui/wk_web_view_configuration_provider.h"
+#import "ios/web/web_state/web_state_impl.h"
+#import "ios/web/webui/web_ui_ios_impl.h"
+#import "net/base/mac/url_conversions.h"
+#include "ui/base/page_transition_types.h"
+#include "url/gurl.h"
+
+#include "components/profile_metrics/browser_profile_type.h"
+
+@interface ChromeWebViewController () <CRWWebStateDelegate,
+                                       CRWWebStateObserver> {
+  raw_ptr<ChromeBrowserState> browser_state_;
+  std::unique_ptr<web::WebState> web_state_;
+  std::unique_ptr<web::WebStateObserverBridge> web_state_observer_;
+  std::unique_ptr<web::WebStateDelegateBridge> web_state_delegate_;
+
+  WKWebView* web_view_;
+  CRWWebController* web_controller_;
```

Comment:
I'm not sure if Obj-C allows this, but these should be `= nullptr;` initialized (including `raw_ptr<ChromeBrowserState> browser_state_ = nullptr;`)

Reply:
Chromium doesn't allow it. If I put that, I get `UnsupportedType` from Chromium when the template compiles:

<img width="914" alt="image" src="https://github.com/brave/brave-core/assets/1530031/0adf41d9-a86d-4e99-894d-14d03982021c">

Also I will get an error compiling because initialization of variables aren't allowed in Objective-C:
```c++
chrome_webview.mm:34:46: error: expected ';' at end of declaration list
   34 |   raw_ptr<ChromeBrowserState> browser_state_ = nullptr;
```

All of these variables are assigned to in the constructor in: `- (instancetype)initWithPrivateBrowsing:(bool)isPrivateBrowsing`